src/test/java/com/teragrep/cfe_39/CombinedFullTest.java

```diff
@@ -0,0 +1,309 @@
+        }
+
+        fs.close();
+    }
+
+    public void hdfsReadCheck() throws IOException {
+        // Check that the files were properly written to HDFS with a read test.
+        String hdfsuri = config.getHdfsuri();
+
+        String path = config.getHdfsPath()+"/"+"testConsumerTopic";
+        // ====== Init HDFS File System Object
+        Configuration conf = new Configuration();
+        // Set FileSystem URI
+        conf.set("fs.defaultFS", hdfsuri);
+        // Because of Maven
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        // Set HADOOP user
+        System.setProperty("HADOOP_USER_NAME", "hdfs");
+        System.setProperty("hadoop.home.dir", "/");
+        //Get the filesystem - HDFS
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+        //==== Create directory if not exists
+        Path workingDir=fs.getWorkingDirectory();
+        Path newDirectoryPath= new Path(path);
+        if(!fs.exists(newDirectoryPath)) {
+            // Create new Directory
+            fs.mkdirs(newDirectoryPath);
+            LOGGER.info("Path {} created.", path);
+        }
+
+        // This is the HDFS write path for the files:
+        // Path hdfswritepath = new Path(newDirectoryPath + "/" + fileName); where newDirectoryPath is config.getHdfsPath() + "/" + lastObject.topic; and filename is lastObject.partition+"."+lastObject.offset;
+
+        // Create the list of files to read from HDFS. Test setup is created so each of the 0-9 partitions will have 2 files with offsets of 8 and 13.
+        List<String> filenameList = new ArrayList<>();
+        for (int i = 0; i <= 9; i++) {
+            filenameList.add(i + "." + 8);
+            filenameList.add(i + "." + 13);
+        }
+        int looper = 0;
+        int partitionCounter = 0;
+        for (String fileName : filenameList) {
+            //==== Read files
+            LOGGER.info("Read file into hdfs");
+            //Create a path
+            Path hdfsreadpath = new Path(newDirectoryPath + "/" + fileName); // The path should be the same that was used in writing the file to HDFS.
+            //Init input stream
+            FSDataInputStream inputStream = fs.open(hdfsreadpath);
+            //The data is in AVRO-format, so it can't be read as a string.
+            DataFileStream<SyslogRecord> reader = new DataFileStream<>(inputStream, new SpecificDatumReader<>(SyslogRecord.class));
+            SyslogRecord record = null;
+            LOGGER.info("\nReading records from file {}:", hdfsreadpath.toString());
+            while (reader.hasNext()) {
+                record = reader.next(record);
+                LOGGER.info(record.toString());
+                // Assert records here like it is done in KafkaConsumerTest.avroReader().
+                if (looper <= 0) {
```

Comment:
This if/else part looks sketchy

Reply:
Refactored in commit 65342c2e865912f981dfe82f470cab6884f72cdb